### PR TITLE
Added CJSON parser, and port to animation API

### PR DESCRIPTION
**Pull request by cryos**
These changes move the animation into the 3Dmol provided API, and
add a custom parser for our Chemical JSON file format. This can be
extended to generate the vibrational mode frames client side, etc.

### CODE REVIEW

**Comment by cryos** [comment]
@cjh1 I would appreciate you looking over this, I think I have left some variables behind that are no longer needed, once this is cleaned up a little more I was going to work on adding the frame generation code client side.

File: app/components/3dmol/3dmol.directive.js [hunk 1]
```diff
@@ -2,6 +2,56 @@ require.ensure(['script!3Dmol/build/3Dmol-min.js'], function(require) {
...
+            "Sn", "Sb", "Te", "I", "Xe", "Cs", "Ba", "La", "Ce", "Pr",
+            "Nd", "Pm", "Sm", "Eu", "Gd", "Tb", "Dy", "Ho", "Er", "Tm",
+            "Yb", "Lu", "Hf", "Ta", "W", "Re", "Os", "Ir", "Pt", "Au",
+            "Hg", "Tl", "Pb", "Bi", "Po", "At", "Rn", "Fr", "Ra", "Ac",
+            "Th", "Pa", "U", "Np", "Pu", "Am", "Cm", "Bk", "Cf", "Es",
+            "Fm", "Md", "No", "Lr", "Rf", "Db", "Sg", "Bh", "Hs", "Mt",
+            "Ds", "Rg", "Cn", "Uut", "Uuq", "Uup", "Uuh", "Uus", "Uuo" ];
+
+        var atoms = [[]];
+        var coords = jsonMol.atoms.coords['3d'];
```
Comment (cjh1) [inline app/components/3dmol/3dmol.directive.js]: I would use dot access here i.e. coords.3d

File: app/components/3dmol/3dmol.directive.js [hunk 2]
```diff
@@ -2,6 +2,56 @@ require.ensure(['script!3Dmol/build/3Dmol-min.js'], function(require) {
...
+            "Zn", "Ga", "Ge", "As", "Se", "Br", "Kr", "Rb", "Sr", "Y",
+            "Zr", "Nb", "Mo", "Tc", "Ru", "Rh", "Pd", "Ag", "Cd", "In",
+            "Sn", "Sb", "Te", "I", "Xe", "Cs", "Ba", "La", "Ce", "Pr",
+            "Nd", "Pm", "Sm", "Eu", "Gd", "Tb", "Dy", "Ho", "Er", "Tm",
+            "Yb", "Lu", "Hf", "Ta", "W", "Re", "Os", "Ir", "Pt", "Au",
+            "Hg", "Tl", "Pb", "Bi", "Po", "At", "Rn", "Fr", "Ra", "Ac",
+            "Th", "Pa", "U", "Np", "Pu", "Am", "Cm", "Bk", "Cf", "Es",
+            "Fm", "Md", "No", "Lr", "Rf", "Db", "Sg", "Bh", "Hs", "Mt",
+            "Ds", "Rg", "Cn", "Uut", "Uuq", "Uup", "Uuh", "Uus", "Uuo" ];
+
```
Comment (cjh1) [inline app/components/3dmol/3dmol.directive.js:21]: " => '

**Comment by cjh1** [comment]
@cryos I have pushed a couple of fixes to tip of this. I also noticed the pause functionality doesn't work. After pausing you can no longer restart the animation, the play button doesn't seem todo anything.

**Comment by cryos** [comment]
This looks good to me, maybe we can merge what we have and make a follow up to fix the pause issues?

**Comment by cjh1** [comment]
sure

**Pull request merged**